### PR TITLE
Add EPSS mirroring to mirror-service

### PR DESCRIPTION
**Pull request by sahibamittal**
Closes https://github.com/DependencyTrack/hyades/issues/926

### CODE REVIEW

File: mirror-service/src/main/java/org/dependencytrack/vulnmirror/datasource/AbstractDatasourceMirror.java [hunk 1]
```diff
@@ -128,6 +130,25 @@ protected void publishIfChanged(final Bom bov) throws ExecutionException, Interr
...
+        }
+        for (EpssItem epssItem : epssItems) {
+            final var serializedEpss = org.dependencytrack.proto.mirror.v1.EpssItem.newBuilder()
+                    .setCve(epssItem.getCve())
+                    .setPercentile(epssItem.getPercentile())
+                    .setEpss(epssItem.getEpss()).build();
+            kafkaProducer.send(new ProducerRecord<>(
+                    KafkaTopic.NEW_EPSS.getName(), epssItem.getCve(), serializedEpss.toByteArray())).get();
+        }
+    }
```
[the remark this answers is not in the document]
Comment (sahibamittal) [reply mirror-service/src/main/java/org/dependencytrack/vulnmirror/datasource/AbstractDatasourceMirror.java:150]: Yes, it should be moved to its specific class EpssMirror 👍🏼

**Comment by nscuro** [comment]
The epss columns will also need to be removed from the `Vulnerability` class, otherwise services will fail to start due to Hibernate's schema validation:

```
vuln-analyzer-1  | __  ____  __  _____   ___  __ ____  ______ 
vuln-analyzer-1  |  --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
vuln-analyzer-1  |  -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
vuln-analyzer-1  | --\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
vuln-analyzer-1  | 2024-04-10 14:11:37,821 ERROR [io.qua.run.Application] (main) Failed to start application (with profile [prod]): java.lang.RuntimeException: Failed to start quarkus
vuln-analyzer-1  |      at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
vuln-analyzer-1  |      at io.quarkus.runtime.Application.start(Application.java:101)
vuln-analyzer-1  |      at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:111)
vuln-analyzer-1  |      at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
vuln-analyzer-1  |      at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
vuln-analyzer-1  |      at io.quarkus.runtime.Quarkus.run(Quarkus.java:124)
vuln-analyzer-1  |      at io.quarkus.runner.GeneratedMain.main(Unknown Source)
vuln-analyzer-1  |      at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
vuln-analyzer-1  |      at java.base/java.lang.reflect.Method.invoke(Method.java:580)
vuln-analyzer-1  |      at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:62)
vuln-analyzer-1  |      at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:33)
vuln-analyzer-1  | Caused by: jakarta.persistence.PersistenceException: [PersistenceUnit: <default>] Unable to build Hibernate SessionFactory
vuln-analyzer-1  |      at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.persistenceException(FastBootEntityManagerFactoryBuilder.java:129)
vuln-analyzer-1  |      at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:89)
vuln-analyzer-1  |      at io.quarkus.hibernate.orm.runtime.FastBootHibernatePersistenceProvider.createEntityManagerFactory(FastBootHibernatePersistenceProvider.java:72)
vuln-analyzer-1  |      at jakarta.persistence.Persistence.createEntityManagerFactory(Persistence.java:80)
vuln-analyzer-1  |      at jakarta.persistence.Persistence.createEntityManagerFactory(Persistence.java:55)
vuln-analyzer-1  |      at io.quarkus.hibernate.orm.runtime.JPAConfig$LazyPersistenceUnit.get(JPAConfig.java:157)
vuln-analyzer-1  |      at io.quarkus.hibernate.orm.runtime.JPAConfig$1.run(JPAConfig.java:64)
vuln-analyzer-1  |      at java.base/java.lang.Thread.run(Thread.java:1583)
vuln-analyzer-1  | Caused by: org.hibernate.tool.schema.spi.SchemaManagementException: Schema-validation: missing column [EPSSPERCENTILE] in table [`VULNERABILITY`]
vuln-analyzer-1  |      at org.hibernate.tool.schema.internal.AbstractSchemaValidator.validateTable(AbstractSchemaValidator.java:145)
...
```

https://github.com/DependencyTrack/hyades/blob/99d70f68f9c8f4a543da9e0c1bd25417408aa706/commons-persistence/src/main/java/org/dependencytrack/persistence/model/Vulnerability.java#L157-L161

**Review by nscuro — COMMENTED** [review]
Once all changes are done, could you please also update the schema we use for tests, so it's consistent with the API server repo?

A script to generate the schema is here: https://github.com/DependencyTrack/hyades-apiserver/blob/main/dev/scripts/dbschema-generate.sh

Location of said schema is here: https://github.com/DependencyTrack/hyades/blob/main/commons-persistence/src/main/resources/schema.sql

Note that this needs to be added to the bottom of the schema file: https://github.com/DependencyTrack/hyades/blob/99d70f68f9c8f4a543da9e0c1bd25417408aa706/commons-persistence/src/main/resources/schema.sql#L1723-L1724

Otherwise tests will start failing.

**Comment by nscuro** [comment]
@sahibamittal Any chance you can address the outstanding Sonar complaints?

**Comment by sahibamittal** [comment]
Sonar complaints addressed but the code is already majorly covered, it is complaining about class registered for reflection which can be avoided here.

**Pull request merged**